.travis.yml

```diff
@@ -1,32 +1,46 @@
 addons:
   hosts:
     - db
+# after_success is only used for Dockerhub and should run last.
 after_success:
-  # Login to dockerhub
-  - echo "$DOCKER_PASSWORD" | docker login -u "$DOCKER_USERNAME" --password-stdin
+  - WEAVIATE_VERSION=$(cat openapi-specs/schema.json| jq -r '.info.version')
   - |
-    # Build & Push docker images on dev and master
-    if [ $TRAVIS_PULL_REQUEST = "false" ]; then
-      if [ $TRAVIS_BRANCH = "master" ]; then
-        # collect the latest version
-        LATESTVERSION=$(git show --oneline HEAD:$WEAVIATESCHEMAFILE | jq -r ".info.version")
-        # Build master
-        DOCKERTARGET=weaviate_base     
-      elif [ $TRAVIS_BRANCH = "develop" ]; then
-        # collect the latest version
-        LATESTVERSION=unstable
-        # Build master
-        DOCKERTARGET=development
-      fi
-      # Build & Push
-      travis_wait 30 docker build -f "Dockerfile-prod" -t creativesoftwarefdn/weaviate:$LATESTVERSION --target $DOCKERTARGET .
-      travis_wait 30 docker push creativesoftwarefdn/weaviate:unstable
+    # only push to Dockerhub if master and not a PR
+    if [ "$TRAVIS_PULL_REQUEST" = "false" ]; then
+        # Login to dockerhub
+        echo "$DOCKER_PASSWORD" | docker login -u "$DOCKER_USERNAME" --password-stdin
+        # prepare release id's
+        DOCKER_ID="creativesoftwarefdn/weaviate:"
+        # releasing a prerelease
+        if [ "$TRAVIS_BRANCH" = "master" ]; then
+            RELEASE_ID=$DOCKER_ID$WEAVIATE_VERSION-$(echo $TRAVIS_COMMIT | cut -c1-5)
```

Comment:
How many digits is that? The git short hash should have 7 digits/characters.

Reply:
better name for `RELEASE_ID` would probably be `TAG` to stay consistent with Docker's domain language.

Reply:
Check